tests/run-make/rustdoc-test-builder/rmake.rs

```diff
@@ -0,0 +1,22 @@
+// This test ensures that if the rustdoc test binary is not executable, it will
+// gracefully fail and not panic.
+
+//@ needs-target-std
+
+use run_make_support::{path, rfs, rustdoc};
+
+fn main() {
+    let absolute_path = path("foo.rs").canonicalize().expect("failed to get absolute path");
+    let output = rustdoc()
+        .input("foo.rs")
+        .arg("--test")
+        .arg("-Zunstable-options")
+        .arg("--test-builder")
+        .arg(&absolute_path)
+        .run_fail();
+
```

Comment:
Lets also assert that the exit code is 1 (ICE gives 101, like other panics), seems more robust than string comparisons.

Reply:
`panic!` isn't always 101 as exit code (depends on the target). So we can't really check for that. ^^'

Discovered that while reading https://github.com/rust-lang/rust/blob/master/library/test/src/test_result.rs#L107.

Reply:
Hum, when we fail doctests, we exit with `101`. ^^'